public/app/core/reducers/navBarTree.ts

```diff
@@ -51,6 +51,29 @@ const navTreeSlice = createSlice({
         }
       }
     },
+    setBookmark: (state, action: PayloadAction<{ item: NavModelItem; isSaved: boolean }>) => {
+      const bookmarks = state.find((navItem) => navItem.id === 'bookmarks');
+      const { item, isSaved } = action.payload;
+      if (bookmarks && config.featureToggles.pinNavItems) {
+        if (isSaved) {
+          if (!bookmarks.children) {
+            bookmarks.children = [];
+          }
+          const newBookmark: NavModelItem = {
+            ...item,
+            // Clear the children and sortWeight of the item
+            children: [],
+            sortWeight: 0,
+          };
+          bookmarks.children.push(newBookmark);
+        } else {
+          const index = bookmarks.children?.findIndex((i) => i.id === item.id) ?? -1;
+          if (index > -1) {
+            bookmarks?.children?.splice(index, 1);
+          }
```

Comment:
```suggestion
         bookmarks.children = bookmarks.children?.filter(i => i.id !== item.id) ?? [];
```
I think this will look simpler with `filter`.